File: second-largest-digit-in-a-string/second-largest-digit-in-a-string.py
class Solution:
    def secondHighest(self, s: str) -> int:
        num = []
        for i in range(len(s)):
            if s[i]>='0' and s[i]<='9':
                num.append(int(s[i]))
        
        num = set(num)
        num = sorted(list(num))

        if len(num)<2:
            return -1
        return num[-2]

File: second-largest-digit-in-a-string/second-largest-digit-in-a-string.cpp
class Solution {
public:
    int secondHighest(string s) {
        vector<int>res ;
        for(int i = 0 ; i < s.length() ; i++){
            if(isdigit(s[i])){
                res.push_back(s[i]-'0');
            }
        }
        unordered_set<int>num(res.begin(),res.end());
        vector<int>ans(num.begin() , num.end());
        sort(ans.begin() , ans.end());
        if (ans.size()<2)
        return -1;
        return ans[ans.size()-2];
    }
};